### PR TITLE
Make Vagrantfiles less hard-coded, and more uniformly configurable

### PR DESCRIPTION
**Pull request by nedbat**
I needed to test a fix in the notifier repo, but could not because NOTIFIER_VERSION was not available in the Vagrantfile.  This makes all the version variables overridable, with less repetition.

### CODE REVIEW

**Comment by nedbat** [comment]
@edx/devops @cpennington Thoughts?  I can make similar changes to other Vagrantfiles once we are done with these.

File: vagrant/base/devstack/Vagrantfile [hunk 1]
```diff
@@ -13,6 +13,22 @@ if ENV["VAGRANT_GUEST_IP"]
   vm_guest_ip = ENV["VAGRANT_GUEST_IP"]
 end
 
+# These are versioning variables in the roles. Each can be overridden, first
+# with OPENEDX_RELEASE, and then with a specific environment variable of the
+# same name but upper-cased.
+VERSION_VARS = [
+    'edx_platform_version',
```
Comment (e0d) [inline vagrant/base/devstack/Vagrantfile:20]: Why not converge format now?

File: vagrant/release/devstack/Vagrantfile [hunk 2]
```diff
@@ -84,6 +68,30 @@ openedx_releases.default = "eucalyptus-devstack-2016-09-01"
...
+    exit 1
+fi
+export PYTHONUNBUFFERED=1
+source /edx/app/edx_ansible/venvs/edx_ansible/bin/activate
+cd /edx/app/edx_ansible/edx_ansible/playbooks
+
+EXTRA_VARS="#{extra_vars_lines}"
+CONFIG_VER="#{rel || 'master'}"
+
+ansible-playbook -i localhost, -c local run_role.yml -e role=edx_ansible -e configuration_version=$CONFIG_VER $EXTRA_VARS
```
Comment (e0d) [inline vagrant/release/devstack/Vagrantfile:92]: Do we remember why two call was necessary here, makes things significantly slower than they could be.

**Review by e0d — COMMENTED** [review]
Generally looks great, with a couple small questions.

File: vagrant/base/devstack/Vagrantfile (hunk 1 above)
Comment (nedbat) [reply vagrant/base/devstack/Vagrantfile:20]: Just a general instinct to make small changes, but I can go through and make them all upper-case if you like.

File: vagrant/release/devstack/Vagrantfile (hunk 2 above)
Comment (nedbat) [reply vagrant/release/devstack/Vagrantfile:92]: I assumed this was the usual two-step: get ansible (etc) right, then use ansible to do all the rest.  Changing this definitely feels outside the scope of this PR.

File: vagrant/release/devstack/Vagrantfile [hunk 3]
```diff
@@ -8,37 +8,21 @@ VAGRANTFILE_API_VERSION = "2"
...
-  CONFIG_VER="master"
-fi
-
-ansible-playbook -i localhost, -c local run_role.yml -e role=edx_ansible -e configuration_version=$CONFIG_VER $EXTRA_VARS
-ansible-playbook -i localhost, -c local vagrant-devstack.yml -e configuration_version=$CONFIG_VER $EXTRA_VARS
-
-SCRIPT
+# These are versioning variables in the roles. Each can be overridden, first
+# with OPENEDX_RELEASE, and then with a specific environment variable of the
+# same name but upper-cased.
```
Comment (cpennington) [inline vagrant/release/devstack/Vagrantfile:13]: It doesn't look like you implemented the per-app variable overrides in this vagrant file. Am I missing something?

Comment (nedbat) [reply vagrant/release/devstack/Vagrantfile:13]: Oh, you are right! It didn't exist at all in the original version, and I didn't add it either! oops.

Comment (nedbat) [reply vagrant/release/devstack/Vagrantfile:13]: Added.

**Review by cpennington — APPROVED** [review]
The casing-gymnastics that you're going through make me wonder whether it would be better to just standardize across the whole repo, but I think it's good to go as-is.

**Comment by nedbat** [comment]
I'll wait on making all the version variables uppercase.  I'd rather move on to the other Vagrantfiles first.

**Pull request merged**